src/internal/agent/hooks/common_test.go

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+package hooks
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/zarf-dev/zarf/src/pkg/state"
+	"github.com/zarf-dev/zarf/src/pkg/transform"
+	"github.com/zarf-dev/zarf/src/test/testutil"
+	"github.com/zarf-dev/zarf/src/types"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/registry/remote"
+)
+
+func populateLocalRegistry(t *testing.T, ctx context.Context, localUrl string, artifact transform.Image) error {
+	localReg, err := remote.NewRegistry(localUrl)
+	if err != nil {
+		return err
+	}
+	localReg.PlainHTTP = true
+
+	remoteReg, err := remote.NewRegistry(artifact.Host)
+	if err != nil {
+		return err
+	}
+
+	src, err := remoteReg.Repository(ctx, artifact.Path)
+	if err != nil {
+		return err
+	}
+	dst, err := localReg.Repository(ctx, artifact.Path)
+	if err != nil {
+		return err
+	}
+	_, err = oras.Copy(ctx, src, artifact.Tag, dst, artifact.Tag, oras.DefaultCopyOptions)
+	if err != nil {
+		return err
+	}
+
+	hashedTag, err := transform.ImageTransformHost(localUrl, fmt.Sprintf("%s/%s:%s", artifact.Host, artifact.Path, artifact.Tag))
+	if err != nil {
+		return err
+	}
+
+	_, err = oras.Copy(ctx, src, artifact.Tag, dst, hashedTag, oras.DefaultCopyOptions)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func setupRegistry(t *testing.T, ctx context.Context) (string, error) {
+	localUrl := testutil.SetupInMemoryRegistry(ctx, t, 5000)
+
+	localReg, err := remote.NewRegistry(localUrl)
+	localReg.PlainHTTP = true
+	if err != nil {
+		return "", err
+	}
+	var artifacts = []transform.Image{
+		{
+			Host: "ghcr.io",
+			Path: "stefanprodan/charts/podinfo",
+			Tag:  "6.9.0",
+		},
+		{
+			Host: "ghcr.io",
+			Path: "stefanprodan/manifests/podinfo",
+			Tag:  "6.9.0",
+		},
+		{
+			Host: "ghcr.io",
+			Path: "stefanprodan/podinfo",
+			Tag:  "6.9.0",
```

Comment:
The chart and manifests should be quick, but pulling down this container image can take a while, try using this container image instead https://github.com/orgs/zarf-dev/packages/container/package/images%2Fhello-world which is super small.